### ISSUE TITLE
Black bar when dragging window to lower DPI monitor

### ISSUE THREAD

**Issue by tomeksowi** [glfw/glfw#1469]
Happens on GLFW 3.3 built with Visual Studio 2017 on Windows 10 Pro.

To reproduce:

1. Run `tests\empty.exe`, window appears on the main monitor (4k, 150%) fine:
![image](https://user-images.githubusercontent.com/583410/56372681-c1f31980-61ff-11e9-8b06-2ce4bdea0ab7.png)
2. Drag window to the secondary monitor (1080p, 100%). Window decoration (title, border) scales down OK:
![image](https://user-images.githubusercontent.com/583410/56373339-029f6280-6201-11e9-9aaf-cb550751e90f.png)
3. Leave the window by dropping on the secondary monitor. Black bar on top and right of the content area appears:
![image](https://user-images.githubusercontent.com/583410/56373467-4a25ee80-6201-11e9-9783-1e4a78f59f95.png)

Other observations:

* It's consistent, I can drag the window back and forth between the monitors and the black bars appear only on the low DPI monitor
* The size of the black bars appears to be the difference of the decoration size (as returned by `glfwGetWindowFrameSize`) between scales 150% and 100%.
* Setting `GLFW_SCALE_TO_MONITOR` does not help.

I'm new to GLFW, I couldn't figure out where's the "content area start point" defined, so far I looked around `win32_window.c`. If someone can tell me where to start looking, I can venture a fix.

**Comment by elmindreda** [comment]
Which version of Windows 10 are you using?

**Comment by tomeksowi** [comment]
@elmindreda, Windows 10 Pro version 1803 build 17134.706.

**Comment by elmindreda** [comment]
That's new enough to have all the DPI scaling bits GLFW uses.

I cannot reproduce this bug on Windows 10 Home 1809. Are you using an Intel GPU? Are the drivers up-to-date? There have been issues before with Intel GPU drivers and Windows 10 (#1069).

**Comment by tomeksowi** [comment]
@elmindreda, yes, Intel HD Graphics 4600. I updated to the latest driver but the problem prevails.

**Comment by elmindreda** [comment]
Right. I'm not sure what GLFW can do to address this. It looks like a driver issue.

Where did you check for latest drivers? Windows Update? Intel? OEM?

**Comment by tomeksowi** [comment]
I checked at [Intel](https://downloadcenter.intel.com/download/28662/Intel-Graphics-Driver-for-Windows-15-40-?product=81496).

How can I prove beyond doubt the problem is caused by the driver and not any layer above it? Any diagnostic/debug/test?

**Comment by elmindreda** [comment]
If you find a non-GLFW OpenGL application that supports Windows 10 non-client DPI scaling then you can see if that has the same problem.

You could try using GLFW 3.2.1, which did things mostly the same way but didn't have non-client DPI scaling support, and see if the offset issue goes away, but that's only half-diagnostic.

**Comment by tomeksowi** [comment]
I tried GLFW 3.2.1, the problem isn't there but this version doesn't have window decoration scaling turned on. But when I added to this piece to `windowProc(…)` to turn it on:

    case WM_NCCREATE:
    {
        EnableNonClientDpiScaling(hWnd);
        break;
    }

Then the black bars predictably appeared in 3.2.1 as well.

I'll try with a non-GLFW app. Any hints on what can make a good test? Something on glut?

**Comment by tomeksowi** [comment]
OK, reproduced the problem with http://www.songho.ca/opengl/files/glWinSimple.zip, just modify the glClearColor call to e.g. blue and add EnableNonClientDpiScaling(hwnd); in procedure.cpp inside "if(msg == WM_NCCREATE)  // Non-Client Create".

Raising to Intel...
